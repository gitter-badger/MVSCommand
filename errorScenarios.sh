#
# Test error scenarios for OSCMD 
#
set -x
mvsjobs --pgm="Greater#8" #name greater than 8 characters
mvsjobs --pgm=Prog --args="parameter that is more than 256 characters......................................................................................................................................................................................................................."
mvsjobs --pgm="~xx"
mvsjobs --pgm="1xx"
mvsjobs --pgm="x~x"
mvsjobs --pgm="xx&"

mvsjobs --sysout=
mvsjobs --sysout=IS#MORE.THAN44.CHARACTR.DATASET.LONGNAME.HERE\(MEMBER\)
mvsjobs --sysout=.
mvsjobs --sysout=TOOLONGQUALIFIER.SRC
mvsjobs --sysout=DATASET\(MEMTOOBIG\) 
mvsjobs --sysout=\(\)
mvsjobs --sysout=A\(
mvsjobs --sysout=A\(\)
