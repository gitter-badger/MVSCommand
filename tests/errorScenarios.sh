#
# Test error scenarios for mvscmd
#
set -x
mvscmd --pgm="Greater#8" #name greater than 8 characters
mvscmd --pgm=Prog --args="parameter that is more than 256 characters......................................................................................................................................................................................................................."
mvscmd --pgm="~xx"
mvscmd --pgm="1xx"
mvscmd --pgm="x~x"
mvscmd --pgm="xx&"

mvscmd --sysout=
mvscmd --sysout=IS#MORE.THAN44.CHARACTR.DATASET.LONGNAME.HERE\(MEMBER\)
mvscmd --sysout=.
mvscmd --sysout=TOOLONGQUALIFIER.SRC
mvscmd --sysout=DATASET\(MEMTOOBIG\)
mvscmd --sysout=\(\)
mvscmd --sysout=A\(
mvscmd --sysout=A\(\)

(mvscmdauth --pgm=RDTD) # 64-bit program from 31-bit mvscmdauth

(mvscmdauth -p=iebcopy) # un-authorized from authorized
(mvscmd -p=ikjeft01) # authorized from unauthorized
