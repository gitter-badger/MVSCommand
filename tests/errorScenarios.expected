+ mvscmd --pgm=Greater#8 
Name Greater#8 is more than 8 characters long.
+ mvscmd --pgm=Prog --args=parameter that is more than 256 characters....................................................................................................................................................................................................................... 
Program arguments more than 100 characters.
+ mvscmd --pgm=~xx 
Name ~xx contains invalid characters.
+ mvscmd --pgm=1xx 
Name 1xx contains invalid characters.
+ mvscmd --pgm=x~x 
Name x~x contains invalid characters.
+ mvscmd --pgm=xx& 
Name xx& contains invalid characters.
+ mvscmd --sysout= 
No dataset name specified for DDName sysout.
+ mvscmd --sysout=IS#MORE.THAN44.CHARACTR.DATASET.LONGNAME.HERE(MEMBER) 
Dataset name IS#MORE.THAN44.CHARACTR.DATASET.LONGNAME.HERE is more than 44 characters long.
+ mvscmd --sysout=. 
Invalid dataset name . specified.
+ mvscmd --sysout=TOOLONGQUALIFIER.SRC 
Name TOOLONGQUALIFIER is more than 8 characters long.
+ mvscmd --sysout=DATASET(MEMTOOBIG) 
Name MEMTOOBIG is more than 8 characters long.
+ mvscmd --sysout=() 
Invalid dataset name () specified.
+ mvscmd --sysout=A( 
Invalid dataset name A( specified.
+ mvscmd --sysout=A() 
No Program name specified with program option.
+ mvscmd --pgm=RDTD 
Error running 64 bit module RDTD with mvscmd (use mvscmd64).
